src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/ZipFileProcessorValidationTest.java

```diff
@@ -0,0 +1,141 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.tasks;
+
+import com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException;
+import org.junit.After;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.springframework.boot.test.rule.OutputCapture;
+import uk.gov.hmcts.reform.bulkscanprocessor.exceptions.DocUploadFailureGenericException;
+import uk.gov.hmcts.reform.bulkscanprocessor.exceptions.FileNameIrregularitiesException;
+import uk.gov.hmcts.reform.bulkscanprocessor.exceptions.NonPdfFileFoundException;
+import uk.gov.hmcts.reform.bulkscanprocessor.services.wrapper.ErrorHandlingWrapper;
+import uk.gov.hmcts.reform.bulkscanprocessor.tasks.processor.ZipFileProcessor;
+import uk.gov.hmcts.reform.bulkscanprocessor.util.EntityParser;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.zip.ZipInputStream;
+
+import static com.google.common.io.Resources.getResource;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowable;
+
+/**
+ * This is unit test. Falls under integration to make use of existing zip file resources.
+ */
+@RunWith(Parameterized.class)
+public class ZipFileProcessorValidationTest {
+
+    @Rule
+    public OutputCapture capture = new OutputCapture();
+
+    private static final Processor processor = new Processor(
+        null,
+        null,
+        null,
+        new ErrorHandlingWrapper((Throwable t) -> {
+            throw new RuntimeException(t);
+        })
+    ) {
+    };
+
+    @Parameterized.Parameters(name = "Processing {0}. Should fail on custom validator? {2}")
+    public static Iterable<Object[]> data() {
```

Comment:
May be it's just me but this now looks bit difficult to read. It would have been nice if we could have created individual test methods for each of the case such as 
1. Scannable items has less entries compared to PDFs.
2. Scannable items has more entries compared to PDFs.
3. Scannable items has same entries but different file names in comparison to PDFs.
I think other scenarios need not be handled by this validation such as can't call dm, no metafile, non-pdf files etc. Those would be handled even before execution comes at this point.